File: factory/src/main/java/com/zuzex/factory/controller/OrderController.java
package com.zuzex.factory.controller;

import com.zuzex.factory.model.Order;
import com.zuzex.factory.service.OrderService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/orders")
public class OrderController {

    private final OrderService orderService;

    @GetMapping
    public List<Order> findAllOrders() {
        return orderService.findAllOrders();
    }

    @GetMapping("/{id}")
    public Order findOrderById(@PathVariable Long id) {
        return orderService.findOrderById(id);
    }

    @PostMapping("/assemble/{orderId}")
    public Order assembleOrder(@PathVariable Long orderId) {
        return orderService.assembleOrderById(orderId);
    }

    @PostMapping("/deliver/{orderId}")
    public Order deliverOrder(@PathVariable Long orderId) {
        return orderService.deliverOrderById(orderId);
    }
}

File: car-factory/src/main/kotlin/com/zuzex/carfactory/controller/OrderController.kt
package com.zuzex.carfactory.controller

import com.zuzex.carfactory.service.OrderService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/orders")
class OrderController(private val orderService: OrderService) {

    @GetMapping
    fun findAllOrders() = orderService.findAllOrders()

    @GetMapping("/{id}")
    fun findOrderById(@PathVariable id: Long) = orderService.findOrderById(id)

    @PostMapping("/action/{id}")
    fun changeOrderStatus(@PathVariable id: Long, @RequestParam action: String) =
        orderService.changeOrderStatusById(id, action)
}
